src/modules/home/screens/carousel-component.tsx

```diff
@@ -0,0 +1,219 @@
+    },
+  },
+  {
+    id: 2,
+    name: 'Computer',
+    price: 36,
+    state: 'New',
+    liked: true,
+    image: {
+      uri: 'https://www.ikea.com/us/en/images/products/stefan-chair-brown-black__0727320_pe735593_s5.jpg?f=s',
+    },
+  },
+  {
+    id: 3,
+    name: 'Computer',
+    price: 36,
+    state: 'New',
+    liked: true,
+    image: {
+      uri: 'https://www.ikea.com/us/en/images/products/stefan-chair-brown-black__0727320_pe735593_s5.jpg?f=s',
+    },
+  },
+  {
+    id: 4,
+    name: 'Computer',
+    price: 36,
+    state: 'New',
+    liked: true,
+    image: {
+      uri: 'https://www.ikea.com/us/en/images/products/stefan-chair-brown-black__0727320_pe735593_s5.jpg?f=s',
+    },
+  },
+  {
+    id: 5,
+    name: 'Another',
+    price: 30,
+    state: 'Restored',
+    liked: true,
+    image: {
+      uri: 'https://www.ikea.com/us/en/images/products/stefan-chair-brown-black__0727320_pe735593_s5.jpg?f=s',
+    },
+  },
+];
+
+type IconType = {
+  focused: boolean;
+};
+
+type CarruselImageType = {
+  url: string;
+};
+
+const Icon = ({ focused, ...props }: IconType) => {
+  const fillColor = focused ? 'red' : 'black';
+
+  return (
+    <View style={[styles.tabIconContainer, focused && styles.tabIconContainerFocused]}>
+      <View style={styles.tabIconWrapper}>
+        <HearthIcon stroke={'none'} fill={fillColor} {...props} />
+      </View>
+    </View>
+  );
+};
+
+const CarouselComponent: React.FC = () => {
+  const renderItem = ({ item }: { item: Product }) => (
+    <View style={styles.itemContainer}>
+      <View
+        style={{
+          alignItems: 'center',
+        }}>
+        <Image source={item.image} style={styles.image} />
+      </View>
+      <View
+        style={{
+          height: 1,
+          backgroundColor: '#e9e8e9',
+          shadowOpacity: 0.4,
+          shadowRadius: 3,
+        }}></View>
+      <View style={styles.priceState}>
+        <Text style={styles.price}>${item.price}</Text>
+        <View style={[styles.textContainer, item.state === 'New' && styles.textContainerNew]}>
+          <Text style={styles.state}>{item.state}</Text>
+        </View>
+      </View>
+      <View style={styles.nameIcon}>
+        <Text style={styles.name}>{item.name}</Text>
+        <Icon focused={item.liked} />
+      </View>
+    </View>
+  );
+
+  return (
+    <View>
+      <Carousel
+        data={products}
+        layout={'default'}
+        renderItem={renderItem}
+        sliderWidth={screenWidth * 1.2}
+        itemWidth={screenWidth * 0.5}
+        inactiveSlideScale={1}
+        inactiveSlideOpacity={1}
+        contentContainerCustomStyle={styles.carouselContentContainer}
+        loop
+      />
+      <View style={{ alignItems: 'center' }}>
+        <Text style={{ color: '#076CE0', fontWeight: 'bold', fontSize: 19 }}>See all</Text>
+      </View>
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
```

Comment:
We prefer having styles on a separate file to have our code better organised and make files more readable :)

Reply:
I'm using tailwind css framework.